app/models/taxon_count.rb

```diff
@@ -8,8 +13,31 @@ class TaxonCount < ApplicationRecord
   TAX_LEVEL_PHYLUM = 6
   TAX_LEVEL_KINGDOM = 7
   TAX_LEVEL_SUPERKINGDOM = 8
+  validates :tax_level, presence: true, inclusion: { in: [
+    TAX_LEVEL_SPECIES,
+    TAX_LEVEL_GENUS,
+    TAX_LEVEL_FAMILY,
+    TAX_LEVEL_ORDER,
+    TAX_LEVEL_CLASS,
+    TAX_LEVEL_PHYLUM,
+    TAX_LEVEL_KINGDOM,
+    TAX_LEVEL_SUPERKINGDOM,
+  ], }, if: :mass_validation_enabled?
+
   COUNT_TYPE_NT = 'NT'.freeze
   COUNT_TYPE_NR = 'NR'.freeze
+  validates :count_type, presence: true, inclusion: { in: [
+    COUNT_TYPE_NT,
+    COUNT_TYPE_NR,
+  ], }, if: :mass_validation_enabled?
+
+  # NOTE: some existing rspec tests assume a value of zero
```

Comment:
Not sure whether to fix the tests or whether zero is a valid count. No zeros exist in prod db.